File: 1421-find-numbers-with-even-number-of-digits/find-numbers-with-even-number-of-digits.java
class Solution {
    public int findNumbers(int[] nums) {
        int count = 0;
        for(int num: nums){
            if(even(num)){
                count++;
            }
        }
        return count;
    }
    static boolean even(int num){
        int no_digits = digits(num);
        return no_digits % 2 == 0;

    }
    static int digits(int num){
        int count = 0;
        while(num > 0){
            num = num/10;
            count++;
        }
        return count;
    }
}

File: 1421-find-numbers-with-even-number-of-digits/find-numbers-with-even-number-of-digits.py
class Solution:
    def findNumbers(self, nums: List[int]) -> int:
        result = []
        count = 0
        for i in nums:
            original = i
            while i > 0:
                i = i//10
                count+=1
            if count%2 == 0:
                result.append(original)
            count = 0
        return len(result)